src/main/kotlin/org/jitsi/rtp/rtcp/rtcpfb/transport_layer_fb/tcc/LastChunk.kt

```diff
@@ -0,0 +1,226 @@
+ * Chrome code as closely as possible in an effort to make
+ * future updates easier.
+ */
+class LastChunk {
+    fun Empty(): Boolean {
+        return size_ == 0
+    }
+
+    fun Clear() {
+        size_ = 0
+        all_same_ = true
+        has_large_delta_ = false
+    }
+
+    // Return if delta sizes still can be encoded into single chunk with added
+    // |delta_size|.
+    fun CanAdd(deltaSize: DeltaSize): Boolean {
+        if (size_ < kMaxTwoBitCapacity)
+            return true
+        if (size_ < kMaxOneBitCapacity && !has_large_delta_ && deltaSize != kLarge)
+            return true
+        if (size_ < kMaxRunLengthCapacity && all_same_ && delta_sizes_[0] == deltaSize)
+            return true
+        return false
+    }
+
+    // Add |delta_size|, assumes |CanAdd(delta_size)|,
+    fun Add(deltaSize: DeltaSize) {
+        if (size_ < kMaxVectorCapacity)
+            delta_sizes_[size_] = deltaSize
+        size_++
+        all_same_ = all_same_ && deltaSize == delta_sizes_[0]
+        has_large_delta_ = has_large_delta_ || deltaSize == kLarge
+    }
+
+    // Encode chunk as large as possible removing encoded delta sizes.
+    // Assume CanAdd() == false for some valid delta_size.
+    fun Emit(): Chunk {
+        if (all_same_) {
+            val chunk = EncodeRunLength()
+            Clear()
+            return chunk
+        }
+        if (size_ == kMaxOneBitCapacity) {
+            val chunk = EncodeOneBit()
+            Clear()
+            return chunk
+        }
+        val chunk = EncodeTwoBit(kMaxTwoBitCapacity)
+        // Remove |kMaxTwoBitCapacity| encoded delta sizes:
+        // Shift remaining delta sizes and recalculate all_same_ && has_large_delta_.
+        size_ -= kMaxTwoBitCapacity
+        all_same_ = true
+        has_large_delta_ = false
+        for (i in 0 until size_) {
+            val deltaSize = delta_sizes_[kMaxTwoBitCapacity + i]
+            delta_sizes_[i] = deltaSize
+            all_same_ = all_same_ && deltaSize == delta_sizes_[0]
+            has_large_delta_ = has_large_delta_ || deltaSize == kLarge
+        }
+
+        return chunk
+    }
+
+    // // Encode all stored delta_sizes into single chunk, pad with 0s if needed.
+    fun EncodeLast(): Chunk {
+        if (all_same_)
+            return EncodeRunLength()
+        if (size_ < kMaxTwoBitCapacity)
+            EncodeTwoBit(size_)
+        return EncodeOneBit()
+    }
+
+    // Decode up to |max_size| delta sizes from |chunk|.
+    fun Decode(chunk: Chunk, max_size: Int) {
+        if ((chunk and 0x8000) == 0) {
+            DecodeRunLength(chunk, max_size)
+        } else if ((chunk and 0x4000) == 0) {
+            DecodeOneBit(chunk, max_size)
+        } else {
+            DecodeTwoBit(chunk, max_size)
+        }
+    }
+
+    // Appends content of the Lastchunk to |deltas|.
+    fun AppendTo(deltas: MutableList<DeltaSize>) {
+        if (all_same_) {
+            deltas.addAll(List(size_) { delta_sizes_[0] })
```

Comment:
i believe it will.  c++ has a call to do something like this so just copied the closest kotlin form of it.  i can change this.  the one below also creates a new list so will change that too